shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/resources/CustomResourceStatus.java

```diff
@@ -63,6 +63,9 @@ public void markConditionFalse(final String conditionType, final String reason,
             condition.get().setStatus(ConditionStatus.False);
             condition.get().setLastTransitionTime(new Date());
             this.conditions.add(condition.get());
+        } else {
```

Comment:
I'm against this personally.

**ALL** `Condition`'s needed to create a resource **MUST** be present from the beginning. The `manager` uses the whole set to determine if a resource is `READY` and adding some in "later" risks the interpretation being wrong if performed before _other_ `Condition`'s are added.